common/components/AddressBookTable.tsx

```diff
@@ -0,0 +1,316 @@
+import React from 'react';
+import { connect, MapStateToProps } from 'react-redux';
+import { AppState } from 'reducers';
+import translate, { translateRaw } from 'translations';
+import {
+  changeAddressLabelEntry,
+  TChangeAddressLabelEntry,
+  saveAddressLabelEntry,
+  TSaveAddressLabelEntry,
+  removeAddressLabelEntry,
+  TRemoveAddressLabelEntry
+} from 'actions/addressBook';
+import {
+  getAddressLabels,
+  getLabelAddresses,
+  getAddressLabelRows,
+  getAddressBookTableEntry
+} from 'selectors/addressBook';
+import { Input, Identicon } from 'components/ui';
+import AddressBookTableRow from './AddressBookTableRow';
+import './AddressBookTable.scss';
+
+interface DispatchProps {
+  changeAddressLabelEntry: TChangeAddressLabelEntry;
+  saveAddressLabelEntry: TSaveAddressLabelEntry;
+  removeAddressLabelEntry: TRemoveAddressLabelEntry;
+}
+
+interface StateProps {
+  rows: ReturnType<typeof getAddressLabelRows>;
+  entry: ReturnType<typeof getAddressBookTableEntry>;
+  addressLabels: ReturnType<typeof getAddressLabels>;
+  labelAddresses: ReturnType<typeof getLabelAddresses>;
+}
+
+type Props = DispatchProps & StateProps;
+
+interface State {
+  editingRow: number | null;
+  addressTouched: boolean;
+  addressBlurred: boolean;
+  labelTouched: boolean;
+  labelBlurred: boolean;
+}
+
+export const ADDRESS_BOOK_TABLE_ID: string = 'ADDRESS_BOOK_TABLE_ID';
+
+class AddressBookTable extends React.Component<Props, State> {
+  public state: State = {
+    editingRow: null,
+    addressTouched: false,
+    addressBlurred: false,
+    labelTouched: false,
+    labelBlurred: false
+  };
+
+  private addressInput: HTMLInputElement | null = null;
+
+  private labelInput: HTMLInputElement | null = null;
+
+  public render() {
+    const {
+      entry: { temporaryAddress = '', addressError = '', temporaryLabel = '', labelError = '' },
+      rows
+    } = this.props;
+    const { addressTouched, addressBlurred, labelTouched, labelBlurred } = this.state;
+
+    // Classnames
+    const addressTouchedWithError = addressTouched && addressError;
+    const labelTouchedWithError = labelTouched && labelError;
+    const addressInputClassName = addressTouchedWithError ? 'invalid' : '';
+    const labelInputClassName = labelTouchedWithError ? 'invalid' : '';
+    const nonMobileTemporaryInputErrorClassName =
+      'AddressBookTable-row-error-temporary-input--non-mobile';
+    const nonMobileVisibleErrorClassName = `${nonMobileTemporaryInputErrorClassName}--visible`;
```

Comment:
For quick little modifiers like these, we have a precedent of using state classes like `is-visible` or `is-non-mobile` to save some line length.

Reply:
@wbobeirne Can you point to an example?

Reply:
https://github.com/MyCryptoHQ/MyCrypto/blob/aa0f9cd455aab2bd7134036fee858de4ae10028c/common/components/NavigationLink.tsx#L28 is an example, here's the SuitCSS guide on the matter https://github.com/suitcss/suit/blob/master/doc/naming-conventions.md#is-stateOfComponent